### ISSUE TITLE
[BUG] Pictures previews not working with chafa and `-f kitty`

### ISSUE THREAD

**Issue by xfzv** [Aloxaf/fzf-tab#303]
I can make sure:
- [x] I am using the latest version of fzf-tab
- [x] this is the minimal zshrc which can reproduce this bug
- [x] fzf-tab is loaded after `compinit`
- [x] fzf-tab is loaded after plugins which will wrap <kbd>Tab</kbd>, like [junegunn/fzf/completion.zsh](https://github.com/junegunn/fzf/blob/master/shell/completion.zsh)
- [x] fzf-tab is loaded before zsh-autosuggestions, zsh-syntax-highlighting and fast-syntax-highlighting.

#### Describe the bug
I followed the wiki for [Previews - show file contents](https://github.com/Aloxaf/fzf-tab/wiki/Preview#show-file-contents).

It works just fine with the following script (taken from the wiki too, slightly modified):

```bash
#! /usr/bin/env bash

has_cmd() { command -v "$1" >/dev/null 2>&1 ; }

MIME=$(file -bL --mime-type "$1")
CATEGORY=${MIME%%/*}
KIND=${MIME##*/}

if [[ -d "$1" ]]; then
    if has_cmd exa; then
        exa -a --color=always -l -g --git --group-directories-first --icons "$1"
    elif has_cmd lsd; then
        lsd -al --color=always --icon=always "$1"
    fi
elif [[ "$CATEGORY" == "image" ]]; then
	if has_cmd chafa; then
        chafa "$1"
    fi

    if has_cmd exiftool; then
        exiftool "$1"
    fi
elif [[ "$CATEGORY" == "text" ]]; then
    if has_cmd bat; then
        bat --color=always --line-range :200 "$1"
    fi
elif [[ "$CATEGORY" == "video" ]]; then
    if has_cmd mediainfo; then
        mediainfo "$1"
    fi
else
    lesspipe.sh "$1" | bat --color=always --line-range :200
fi
```

Relevant part of my `fzf-tab` config:

```zsh
zstyle ':fzf-tab:complete:*:*' fzf-preview 'less ${(Q)realpath}'
zstyle ':fzf-tab:complete:*:*' fzf-flags --height=100% --preview-window=right:wrap
zstyle ':fzf-tab:complete:*:options' fzf-preview
zstyle ':fzf-tab:complete:*:argument-1' fzf-preview
export LESSOPEN='|/path/to/lessfilter.sh %s'
```

However, pictures previews aren't working with `chafa` when using kitty terminal (and default preview format).

From `man chafa`:

```
       -f, --format format
           Set output format; one of [iterm, kitty, sixels, symbols]. The default is iterm, kitty or sixels if the connected terminal
           supports one of these, falling back to symbols ("ANSI art") otherwise.
```

Since I'm using kitty, running `chafa` without any argument defaults to `-f kitty`, which doesn't seem to work:

![image](https://user-images.githubusercontent.com/78810647/184022668-50360e15-a216-4748-882b-a6321ff97b75.png)

It works fine if run directly in terminal with:

```zsh
❯ chafa 01.jpg
```

![image](https://user-images.githubusercontent.com/78810647/184022713-eb1362c7-ccd0-46af-8bb1-775b83a3ed34.png)

If using

```zsh
chafa -f symbols "$1"
```

in the script, it works too, but the preview quality is very bad of course:

![image](https://user-images.githubusercontent.com/78810647/184022728-1b76188c-6008-4119-98ce-25c81db8036d.png)

#### Environment:
 - OS: Gentoo

```zsh
❯  zsh --version
zsh 5.9 (x86_64-pc-linux-gnu)

❯  kitty --version
kitty 0.25.2 created by Kovid Goyal

❯  chafa --version
Chafa version 1.12.3
```

Am I missing something or is this a fzf limitation?

Thanks.

**Comment by xfzv** [comment]
I've been playing around with [ytfzf](https://github.com/pystardust/ytfzf):

- Above:
```zsh
% ytfzf -t --thumb-viewer=kitty Gentoo
```
- Below:
```zsh
% ytfzf -t --thumb-viewer=chafa Gentoo
```

![image](https://user-images.githubusercontent.com/78810647/184108109-d5f6e4c1-32c6-455a-8b44-f7ba588bd1a1.png)

I tried to replace [this line](https://github.com/pystardust/ytfzf/blob/32d69eb04c7059ff246ae8e9dc638f36ed49600b/ytfzf#L1752-L1755) from `/usr/bin/ytfzf`:

```diff
chafa)
    printf '\n'
    command_exists "chafa" || die 3 "\nchafa is not installed\n"
-   chafa --format=symbols -s "$((width-4))x$height" "$thumb_path" ;;
+   chafa --format=kitty -s "$((width-4))x$height" "$thumb_path" ;;
```

and got the exact same issue (wall of random characters instead of image preview) when using `ytfzf -t --thumb-view=chafa`.

I also noticed [these lines](https://github.com/pystardust/ytfzf/blob/32d69eb04c7059ff246ae8e9dc638f36ed49600b/ytfzf#L1778-L1785) and somewhat got it working by replacing (in my `lessfilter.sh` script)

```diff
elif [[ "$CATEGORY" == "image" ]]; then
-    has_cmd chafa && chafa "$1"
+    kitty +kitten icat --clear --transfer-mode file
+    kitty +kitten icat --transfer-mode file "$1"
    has_cmd exiftool && exiftool "$1"
```

but of course it's very buggy since I removed the `--place` option and maybe the other options aren't optimal either. The image is displayed full size instead of matching fzf preview window size and the preview stays displayed when moving with arrow keys while in fzf.

So based on what I'm seeing with `ytfzf`, I guess `chafa` doesn't play well with `fzf` and `--format kitty`. The best way would be to mimic `ytfzf` implementation with ` kitty +kitten icat` and `ueberzug` to determine thumbnail position and size.

Maybe someone could help in that regard.

**Comment by justchokingaround** [comment]
any updates on this?

**Comment by justchokingaround** [comment]
yes, i actually saw the announcement, however even after building from source, the example code didn't seem to work for me in the kitty terminal

**Comment by justchokingaround** [comment]
oh nice, thank you, i'll check another day, but this is pretty cool. i think it's a big step forward that fzf has native image support, hopefully this means sixel is possible in the future :pray:

**Comment by DivineMK** [comment]
[fzf now supports image ](https://github.com/junegunn/fzf?tab=readme-ov-file#previewing-an-image)

**Comment by DivineMK** [comment]
May I see the whole .lessfilter? Maybe you happen to have other commands for images like chafa as well?

**Comment by DivineMK** [comment]
From [lesspipe documentation](https://github.com/wofr06/lesspipe?tab=readme-ov-file#9-user-defined-filtering) you need to use .lessfilter or lessfilter not lessfilter.sh.

**Comment by DivineMK** [comment]
If running fzf-preview.sh in terminal works then I don't know how to help :cry:

**Comment by Chrispycode** [comment]
Is this still an Issue with kitty, I got it to work with previews like this:
https://gist.github.com/Chrispycode/b9a3180ed5f01c4df4c765221d03b8b3

[Screencast from 2024-05-21 20-00-40.webm](https://github.com/Aloxaf/fzf-tab/assets/22537998/1f33f0fe-ecbf-483f-ad36-cd3cfb59ed28)

**Comment by Chrispycode** [comment]
The fzf-preview.sh also only works if I add `>&2` and then it also returns the message but beneath the image.

```
elif [ "$category" = image ]; then
  # kitty icat --silent --clear --transfer-mode=memory --stdin=no --place="$dim@0x0" "$1" >&2
  "$DOTS/fzf-preview.sh" "$1" >&2
```

![Screenshot from 2024-05-21 23-07-51](https://github.com/Aloxaf/fzf-tab/assets/22537998/128d6a38-5225-45c2-861c-17a29fa07898)

**Comment by Chrispycode** [comment]
I got it to work now without the message, seems to be a problem with the LESSOPEN example.

```
zstyle ':fzf-tab:complete:*:*' fzf-preview '$DOTS/fzf-preview.sh ${(Q)realpath}'
```

**Comment by Chrispycode** [comment]
yes I only replaced the first case in the fzf-preview.sh for the preview of directories with 

```
if [[ ! $type =~ image/ ]]; then
  if [[ $type =~ =binary ]]; then
    ls -1lah --color "$1"
    exit
  fi
```
